sdk/src/main/scala/com/horizen/certificatesubmitter/CertificateSubmitter.scala

```diff
@@ -196,11 +200,19 @@ class CertificateSubmitter
     }
   }
 
+  private def getCertificateTopQuality(): Int = {
+    mainchainChannel.getTopQualityCertificates(BytesUtils.toHexString(BytesUtils.reverseBytes(params.sidechainId))) match {
+      case Success(topQualityCertificates)=>
+        topQualityCertificates.mempoolCertInfo.quality.getOrElse(topQualityCertificates.chainCertInfo.quality.getOrElse(0))
```

Comment:
ignore from block certificates the one not belonging to the expected epoch and assert that the one in the mempool is of the current epoch